### PR TITLE
Remove `IncludeCategories` from `.clang-format`

### PR DESCRIPTION
**Pull request by codereport**
It was recently noticed that the `IncludeCategories`:
```
IncludeCategories:
  - Regex:           '^<ext/.*\.h>'
    Priority:        2
  - Regex:           '^<.*\.h>'
    Priority:        1
  - Regex:           '^<.*'
    Priority:        2
  - Regex:           '.*'
    Priority:        3
```
In the `.clang-format` are not really necessary as `ext` has no meaning in RAPIDS. This PR removes these.

Depends on https://github.com/rapidsai/cudf/pull/9870.

### CODE REVIEW

**Comment by codereport** [comment]
Note CI is failing because of https://github.com/rapidsai/cudf/pull/9870 not because of the clang-format changes

**Review by ttnghia — APPROVED** [review]

**Review by bdice — APPROVED** [review]
clang-format changes look good. Part of me wants to fix the `#include "..."`s that should be using `<...>` revealed in this changeset, but that's not a blocker since the changes were automated and this PR touches a lot of files that would be time consuming to fix.

File: cpp/tests/binaryop/binop-compiled-test.cpp [hunk 1]
```diff
@@ -26,9 +26,9 @@
 #include <cudf_test/column_wrapper.hpp>
 #include <cudf_test/type_lists.hpp>
 
+#include "cudf/utilities/error.hpp"
```
Comment (mythrocks) [inline cpp/tests/binaryop/binop-compiled-test.cpp:29]: Should this not have been the following?
```suggestion
#include <cudf/utilities/error.hpp>
```

File: cpp/tests/bitmask/bitmask_tests.cpp [hunk 2]
```diff
@@ -24,9 +24,9 @@
 #include <cudf_test/column_wrapper.hpp>
 #include <cudf_test/cudf_gtest.hpp>
 
-#include <rmm/device_buffer.hpp>
 #include "rmm/cuda_stream_view.hpp"
 #include "rmm/device_uvector.hpp"
```
Comment (mythrocks) [inline cpp/tests/bitmask/bitmask_tests.cpp:28]:
```suggestion
#include <rmm/cuda_stream_view.hpp>
#include <rmm/device_uvector.hpp>
```

File: cpp/tests/partitioning/hash_partition_test.cpp [hunk 3]
```diff
@@ -13,6 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#include "cudf/detail/utilities/vector_factories.hpp"
```
Comment (mythrocks) [inline cpp/tests/partitioning/hash_partition_test.cpp:16]:
```suggestion
#include <cudf/detail/utilities/vector_factories.hpp>
```

File: cpp/tests/replace/replace_tests.cpp [hunk 4]
```diff
@@ -31,11 +31,11 @@
 
 #include <thrust/iterator/transform_iterator.h>
 
-#include <gtest/gtest.h>
+#include "cudf/types.hpp"
```
Comment (mythrocks) [inline cpp/tests/replace/replace_tests.cpp:34]:
```suggestion
#include <cudf/types.hpp>
```

File: cpp/tests/reshape/tile_tests.cpp [hunk 5]
```diff
@@ -20,9 +20,9 @@
 #include <cudf_test/table_utilities.hpp>
 #include <cudf_test/type_lists.hpp>
 
+#include "cudf/utilities/error.hpp"
```
Comment (mythrocks) [inline cpp/tests/reshape/tile_tests.cpp:23]:
```suggestion
#include <cudf/utilities/error.hpp>
```

File: cpp/tests/scalar/scalar_device_view_test.cu [hunk 6]
```diff
@@ -25,9 +25,9 @@
 #include <cudf_test/type_list_utilities.hpp>
 #include <cudf_test/type_lists.hpp>
 
-#include <thrust/sequence.h>
-#include <random>
 #include "rmm/cuda_stream_view.hpp"
```
Comment (mythrocks) [inline cpp/tests/scalar/scalar_device_view_test.cu:28]:
```suggestion
#include <rmm/cuda_stream_view.hpp>
```

**Review by mythrocks — APPROVED** [review]
+1 for this change, save for keeping the `#include "header.hpp"` style of includes.
Please ignore if that's not the focus of changes here. I've flagged them as `suggestion`s where I could, for easy correction.

**Review by rgsl888prabhu — APPROVED** [review]

File: cpp/tests/binaryop/binop-compiled-test.cpp (hunk 1 above)
Comment (mythrocks) [reply cpp/tests/binaryop/binop-compiled-test.cpp:29]: Actually, no. This can't be committed without also moving this to line#23.

**Comment by codereport** [comment]
> +1 for this change, save for keeping the `#include "header.hpp"` style of includes. Please ignore if that's not the focus of changes here. I've flagged them as `suggestion`s where I could, for easy correction.

It's all good. I committed them all, pulled down and reran clang-format. Thanks :+1:

**Comment by codereport** [comment]
rerun tests

**Comment by codereport** [comment]
rerun tests

**Comment by codereport** [comment]
rerun tests

**Comment by codereport** [comment]
@gpucibot merge

**Pull request merged**